src/main/java/com/sforce/async/BulkConnection.java

```diff
@@ -801,16 +802,13 @@ static void serializeToJson(OutputStream out, Object value) throws IOException{
 
     /**
      * Deserialize JSON input
-     * @param in
-     * @param tmpClass
-     * @param <T>
-     * @return
-     * @throws IOException
-     * @throws ConnectionException
      */
-    static <T> T deserializeJsonToObject (InputStream in, Class<T> tmpClass) throws IOException, ConnectionException {
+    static <T> T deserializeJsonToObject (InputStream in, Class<T> tmpClass) throws IOException {
         ObjectMapper mapper = new ObjectMapper();
         mapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+        // By default, ObjectMapper generates Calendar instances with UTC TimeZone.
+        // Here, override that to "GMT" to better match the behavior of the WSC XML parser.
+        mapper.setTimeZone(TimeZone.getTimeZone("GMT"));
```

Comment:
https://salesforce.quip.com/ayWtAMwWqBjL#FQFACAPGwx5 explains XML Parser behavior is setting the timezone to default if the POST response doesn't have 'Z' appended to it. I think instead of explicitly setting the timezone to 'GMT', we should use [getDefault()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TimeZone.html#getDefault()) method to set it to default timezone of JVM which is similar to behavior of older version of Jackson.